src/evergreen/api.py

```diff
@@ -61,9 +61,22 @@
 
 CACHE_SIZE = 5000
 DEFAULT_LIMIT = 100
-MAX_RETRIES = 3
-START_WAIT_TIME_SEC = 2
-MAX_WAIT_TIME_SEC = 5
+
+DEFAULT_HTTP_RETRY_ATTEMPTS = 10
+DEFAULT_HTTP_RETRY_BACKOFF_FACTOR = 0.1
+DEFAULT_HTTP_RETRY_BACKOFF_MAX_SEC = 120
+DEFAULT_HTTP_RETRY_CODES = frozenset(
```

Comment:
This list came from some investigation I did... in particular, I used the following references:
 - https://stackoverflow.com/questions/51770071/what-are-the-http-codes-to-automatically-retry-the-request
 - https://medium.com/@API4AI/best-practice-implementing-retry-logic-in-http-api-clients-0b5469c08ced
 - https://github.com/sindresorhus/got/blob/4a44fc40439003e94975e6425b021545b0faaa9a/documentation/7-retry.md#retry-api
 
 I feel like we could theoretically retry all 5xx errors. Since this are all server side error. while for the 4xx errors, we should retry only a subset of them.